cloud/aws/providerconfig/types.go

```diff
@@ -14,25 +14,132 @@
 package providerconfig
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AWSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderConfig field
+// for an AWS instance.
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 type AWSMachineProviderConfig struct {
-	metav1.TypeMeta `json:",inline"`
+	metav1.TypeMeta
+
+	// AMI is the reference to the AMI from which to create the machine instance.
+	AMI AWSResourceReference
+
+	// InstanceType is the type of instance to create. Example: m4.xlarge
+	InstanceType string
+
+	// Tags is the set of tags to add to apply to an instance, in addition to the ones
+	// added by default by the actuator. These tags are additive. The actuator will ensure
+	// these tags are present, but will not remove any other tags that may exist on the
+	// instance.
+	Tags []TagSpecification
```

Comment:
Isn't it better for this type to be `map[string]string`?
AWS client accepts `map[string]string` already and with this implementation, you have to loop over every single item to put them in a `map`

Reply:
The only reason I did it this way is: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#lists-of-named-subobjects-preferred-over-maps
But if there are no objections to reverting it back to a map, then I can do so.

Reply:
Actually now that I read the guidelines again... this is a pure map ... not a map of subobjects, so it doesn't make sense to make it an array.